eth/downloader/peer.go

```diff
@@ -305,13 +311,42 @@ func (p *peer) Demote() {
 	}
 }
 
+// Ignore appends a new entity to the set of items (blocks, receipts, states)
+// that a peer is known not to have (i.e. have been requested before). If the
+// set reaches its maximum allowed capacity, items are randomly dropped off.
+func (p *peer) Ignore(hash common.Hash) {
+	p.ignoreLock.Lock()
+	defer p.ignoreLock.Unlock()
+
+	for len(p.ignored) >= maxIgnoredHashes {
+		for drop, _ := range p.ignored {
+			delete(p.ignored, drop)
+			break
+		}
+	}
+	p.ignored[hash] = struct{}{}
+}
+
+// Ignored retrieves whether the hash of a blockchain item is on the peers ignore
+// list or not (i.e. whether we know that the peer does not have it).
```

Comment:
Hm, can't argue with personal style. I just don't like stuff that looks like it is public but isn't.
Any idea about the name?